src/components/MapSettings/TimeSelector.js

```diff
@@ -53,16 +50,18 @@ export default class TimeSelector extends React.Component {
       return <NullTimeSelector/>;
     }
 
+    const label =
+      <TimeSelectorLabel temporalLabelPart={this.temporalLabelPart()}/>
+    ;
```

Comment:
Possibly inline  `this.temporalLabelPart()`. It doesn't really serve any purpose being pulled out as a function: it's evaluated once, here.